### ISSUE TITLE
Support max_surge in non-beta GKE cluster variants

### ISSUE THREAD

**Issue by catherinetcai** [terraform-google-modules/terraform-google-kubernetes-engine#1258]
### TL;DR

Surge upgrades for GKE are supported [in the beta cluster variants](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/394). Would it be possible to add support for the max_surge upgrades in the non-beta cluster variants as well? 

### Terraform Resources

We just need to expose the [upgrade_settings section](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/container_node_pool#upgrade_settings) to the private cluster variants.

### Detailed design

We just need to take [this section](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/81686e7ffb98da59c756691351db1e8ae158c218/modules/beta-private-cluster-update-variant/cluster.tf#L502-L505) and add it [here](https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/81686e7ffb98da59c756691351db1e8ae158c218/modules/private-cluster/cluster.tf#L242).

**Comment by bharathkkb** [comment]
Hi @catherinetcai 
Happy to review a PR adding this. We can remove the beta feature gate here.
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/81686e7ffb98da59c756691351db1e8ae158c218/autogen/main/cluster.tf.tmpl#L575
